src/chapters/aspdotnet-controllers-and-routes/controllers-attributes.rst

```diff
@@ -59,44 +63,44 @@ Since we want to map ``Index()`` to the path ``localhost:5001/helloworld``, we r
 If we hadn't, we would have inadvertently mapped the ``Index()`` method to the path, ``localhost:5001/helloworld/helloworld``.
 We want to leave the ``[HttpGet]`` attribute above the ``Index()`` method so we can still specify that ``Index()`` responds to ``GET`` requests.
 
-Now that just leaves us the ``Welcome()`` and ``Display()`` methods!
+Now that just leaves us the ``Welcome()`` method!
 
 One Method, Two Request Types
------------------------------
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Despite ``Welcome()`` and ``Display()`` responding to different request types, the two methods return very similar strings of HTML.
+When we modified the ``Welcome()`` method to respond to a ``POST`` request, we commented out the attributes that we added when we were working with route parameters.
 With attributes, we can DRY our code and create one method that can respond to two different request types at two different routes.
 Before we begin, we should note that we can add route info directly to ``[HttpGet]`` and ``[HttpPost]``.
 
 .. sourcecode:: csharp
    :linenos:
 
-   [HttpPost("display")]
-   public IActionResult Display(string name = "World")
+   [HttpPost("welcome")]
+   public IActionResult Welcome(string name = "World")
    {
-      return Content("<h1>Hello " + name + "!</h1>", "text/html");
+      return Content("<h1>Welcome to my app, " + name + "!</h1>", "text/html");
    }
 
 On line 1, we modified the ``[HttpPost]`` attribute to include the end of our path.
-Now ``Display()`` still responds to ``POST`` requests at ``localhost:5001/helloworld/display``.
+Now ``Welcome()`` still responds to ``POST`` requests at ``localhost:5001/helloworld/welcome``.
 However, this is just the beginning of us DRYing our code.
 
-We want to comment out the ``Welcome()`` method and add another attribute to ``Display()`` so that we have one method responding to two different types of HTTP requests at two different paths.
-We can add an ``[HttpGet]`` attribute to ``Display()`` to do so.
+We also want ``Welcome()`` to respond to ``GET`` requests.
+We can modify an ``[HttpGet]`` attribute to do so.
 
 .. sourcecode:: csharp
    :linenos:
 
    [HttpGet("welcome/{name?}")]
-   [HttpPost("display")]
-   public IActionResult Display(string name = "World")
+   [HttpPost("welcome")]
```

Comment:
im still a little concerned that this example kind of puts a lot of ideas together. will the students confidently grasp which request type is used when the form is submitted? is this indicative of a real app scenario they will eventually encounter?

Reply:
I changed up the route and double-checked the info we provide them on HTTP to make sure it is clear why we use POST. I think the students got it!